database/seeders/ResourceSeeder.php

```diff
@@ -18,63 +19,84 @@ class ResourceSeeder extends Seeder
-                ],
-                'phases' => ['design'],
-                'type' => ContentType::firstWhere('name->en', 'Guidelines and best practices'),
-                'sectors' => [Sector::firstWhere('name->en', 'Federal government programs and services')->id],
-                'impacts' => [Impact::firstWhere('name->en', 'Programs and services')->id, Impact::firstWhere('name->en', 'Communication, other than information and communication technologies')->id],
-                'topics' => [],
-                'resourceCollections' => [ResourceCollection::firstWhere('title->en', 'The Accessible Canada Act')->id],
-            ],
-            [
-                'title' => ['en' => 'An Introduction To The Accessible Canada Act'],
-                'author' => ['en' => 'ARCH Disability Law Centre'],
-                'url' => ['en' => 'https://archdisabilitylaw.ca/an-introduction-to-the-accessible-canada-act/'],
-                'phases' => ['design'],
-                'sectors' => [Sector::firstWhere('name->en', 'Federal government programs and services')->id],
-                'impacts' => [Impact::firstWhere('name->en', 'Programs and services')->id, Impact::firstWhere('name->en', 'Communication, other than information and communication technologies')->id],
-                'topics' => [],
-                'resourceCollections' => [ResourceCollection::firstWhere('title->en', 'The Accessible Canada Act')->id],
-            ],
-        ];
 
-        foreach ($resources as $resource) {
-            $item = Resource::firstOrCreate([
-                'title' => $resource['title'],
-                'author' => $resource['author'] ?? '',
-                'summary' => $resource['summary'] ?? '',
-                'url' => $resource['url'],
-                'phases' => $resource['phases'] ?? [],
-            ]);
+        // check if there is a JSON file that has stored data for the seeder
+        if (Storage::disk('seeds')->exists('resources.json')) {
+            $resources = json_decode(Storage::disk('seeds')->get('resources.json'), true);
 
-            if (isset($resource['type'])) {
-                $item->contentType()->associate($resource['type']);
-                $item->save();
+            foreach ($resources as $resource) {
+                $item = Resource::firstOrCreate([
+                    'title' => json_decode($resource['title'], true),
+                    'slug' => json_decode($resource['slug'], true),
+                    'summary' => json_decode($resource['summary'], true),
+                    'url' => json_decode($resource['url'], true),
+                    'formats' => json_decode($resource['formats'], true),
+                    'phases' => json_decode($resource['phases'], true),
+                    'author' => json_decode($resource['author'], true),
+                    'content_type_id' => $resource['content_type_id'],
+                    'organization_id' => $resource['organization_id'],
```

Comment:
Here is another option. We just skip those lines and show the error so it can be fixed if the user is restoring at the command line. Here is the testing output that I ran a test on.

```bash

 Which class would you like to restore? [InterpretationSeeder]:
  [0] IdentitySeeder
  [1] InterpretationSeeder
  [2] ResourceCollectionSeeder
  [3] ResourceSeeder
  [4] TopicSeeder
 > 3

Running seeder ResourceSeeder
   INFO  Seeding database.  

Skipping insert because content type with id 1 is missing 
Skipping insert because content type with id 1 is missing 
Skipping insert because content type with id 1 is missing 
```

Reply:
Let me know if you want me to keep this? We can also remove these from the configuration for tables and classes in the short term and come back for more solid solutions in the future.

Reply:
I guess that's fine for this case. So we restore only the ones that don't have missing foreign keys?

Reply:
Yes we just make each seeder robust to missing foreign keys if they have them.